File: app/src/main/java/com/art2cat/dev/moonlightnote/Utils/ToastUtils.java
package com.art2cat.dev.moonlightnote.utils;

import android.content.Context;
import android.widget.Toast;

/**
 * Created by Rorschach
 * on 2017/1/13 10:52.
 */

public class ToastUtils {


    private Context context;
    private String content;

    private static ToastUtils newInstance() {
        return new ToastUtils();
    }
    public static ToastUtils with(Context context) {
        ToastUtils toastUtils = ToastUtils.newInstance();
        toastUtils.setContext(context);
        return toastUtils;
    }

    public ToastUtils setMessage(String message) {
        this.setContent(message);
        return this;
    }

    public void showShortToast() {
        Toast.makeText(context, content, Toast.LENGTH_SHORT).show();
    }

    public void showLongToast() {
        Toast.makeText(context, content, Toast.LENGTH_LONG).show();
    }

    private void setContext(Context context) {
        this.context = context;
    }

    private void setContent(String message) {
        this.content = message;
    }
}

File: app/src/main/java/com/art2cat/dev/moonlightnote/utils/ToastUtils.kt
package com.art2cat.dev.moonlightnote.utils

import android.content.Context
import android.widget.Toast

/**
 * Created by Rorschach
 * on 2017/1/13 10:52.
 */

open class ToastUtils {


    private var context: Context? = null
    private var content: String? = null

    fun setMessage(message: String): ToastUtils {
        this.setContent(message)
        return this
    }

    fun showShortToast() {
        Toast.makeText(context, content, Toast.LENGTH_SHORT).show()
    }

    fun showLongToast() {
        Toast.makeText(context, content, Toast.LENGTH_LONG).show()
    }

    private fun setContent(message: String) {
        this.content = message
    }

    companion object {


        fun with(context: Context): ToastUtils {
            Instance.toastUtils.context = context
            return Instance.toastUtils
        }
    }

    object Instance {
        val toastUtils = ToastUtils()
    }
}
